Lib/functools.py

```diff
@@ -236,7 +236,7 @@ def __ge__(self, other):
 
 def reduce(function, sequence, initial=_initial_missing):
     """
-    reduce(function, iterable[, initial], /) -> value
+    reduce(function, iterable, /, initial=None) -> value
```

Comment:
Maybe use ellipsis:

```suggestion
    reduce(function, iterable, /, initial=...) -> value
```

Reply:
But the sentinel is private and doesn't even exist in the C implementation and ellipsis is frequently used unspecified default values in typeshed. We could use multiple signatures though.

Reply:
> But the sentinel is private and doesn't even exist in the C implementation.

It's easy to add, see https://github.com/python/cpython/pull/125917#issuecomment-2434922118

> Ellipsis is frequently used for unspecified default values in typeshed.

@Eclips4?

> We could use multiple signatures though.

Yes, I think it's fine for the sphinx docs.  But help will looks like this (as for pure-Python version):
```pycon
>>> help(functools.reduce)
Help on built-in function reduce in module _functools:

reduce(function, iterable, /,
       initial=_functools._initial_missing)
    Apply a function of two arguments cumulatively to an iterable, from left to right.

    [...]
```

Reply:
I don't think I've ever seen `=...` in the docs. Do we have precedent for that?

Reply:
> But the sentinel is private and doesn't even exist in the C implementation. Ellipsis is frequently used for unspecified default values in typeshed. We could use multiple signatures though.


Multiple signatures for a docs sounds like a good solution.
Using `...` for default values is essentially the same as using `None`, and it's just wrong since users can pass `...` as the `initial` value.

Reply:
> I don't think I've ever seen =... in the docs. Do we have precedent for that?

Yeah, e.g. for the int.from_bytes, for example.

> it seems like the signature is giving inspect a hard time. But it is autogenerated by AC. Did I do something wrong?

First, note that reduce() has no correct signature in the current main.

Now AC adds one, but it can't be parsed by ``inspect._signature_fromstr()``: this helper has own opinion on what can be specified as a default value (e.g. it can't be a complex number).